pkg/patch/patch.go

```diff
@@ -337,6 +346,36 @@ func patchWithContext(ctx context.Context, ch chan error, image, reportFile, pat
 	return err
 }
 
+func getExporterType(ctx context.Context, imageName, ref string) (string, error) {
+	defaultExporter := client.ExporterDocker
+	repo, err := remote.NewRepository(imageName)
```

Comment:
I'm not sure if this library would work for local/private images?

Reply:
ah I missed that, thank you for pointing that out @ashnamehrotra!
the code should now handle both local and private in addition to public remotes.
however, I am not sure if it's the right approach, I initially tried to use build kit via `ResolveImageConfig`, but it doesn't appear to expose the media type. it's possible i could have missed it